action.yml

```diff
@@ -91,6 +94,14 @@ runs:
             --token ${{ steps.get-registration.outputs.registration_token }} --name $(hostname) --work /home/ghrunner \
             --labels ${{ steps.get-registration.outputs.runner_scope }} --ephemeral
 
+          # Running custom user data if provided
+          if [[ -n "${{ inputs.custom_user_data }}" ]]; then
+            echo "Running custom user data"
+            echo "${{ inputs.custom_user_data }}" | tee /tmp/custom_user_data.sh
+            chmod +x /tmp/custom_user_data.sh
+            sudo /tmp/custom_user_data.sh
```

Comment:
`sudo` changes user to `root` is that intentional? 
E.g. if I want to run `pip install --user foobar` it will be installed in `/root` instead of current user.

Reply:
@sthaha @vprashar2929 cloud-init is run as the root user. `sudo` can be removed.